src/app/products_view/product_availability_details/product_availability_details.ts

```diff
@@ -0,0 +1,23 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { ProductQuantityInfo } from '../../shared/model/product_quantity_info';
+
+@Component({
+  selector: 'tk-product-availability-details',
+  templateUrl: './product_availability_details.ng.html',
+  styleUrls: ['./product_availability_details.scss'],
+})
+export class ProductAvailabilityDetails implements OnInit {
+  @Input() productQuantityInfo: ProductQuantityInfo[] = [];
+
+  selectedCountry: ProductQuantityInfo;
+
+  constructor() {}
+
+  ngOnInit() {
+    this.selectedCountry = this.productQuantityInfo[0];
+  }
+
+  onCountryChange(countryName: ProductQuantityInfo) {
```

Comment:
```suggestion
  onCountryChange(country: ProductQuantityInfo) {
```
Argument name is a bit confusing. Passed data represents more than a country name.